src/lib/machineAdjustments.ts

```diff
@@ -0,0 +1,286 @@
+      alias: 'Stop',
+      required: false
+    },
+    leftPosition: {
+      range: ['1', '2', '3', '4', '5', '6', '7', '8'],
+      alias: 'Leg',
+      required: false
+    },
+    seat: {
+      range: ['1', '2', '3', '4', '5', '6', '7', '8', '9', '0', 'A', 'C', 'E', 'F', 'H', 'J'],
+      alias: 'Back',
+      required: false
+    }
+  },
+  '001299': { /* Leg Curl Pro */ },
+  '001598': { // Squat Pro
+    leftPosition: {
+      range: ['1', '2', '3', '4', '5', '6', '7'],
+      alias: 'Shoulder Pads',
+      required: false
+    }
+  },
+  '001699': { // Military Press
+    seat: {
+      range: ['1', '2', '3', '4', '5', '6', '7', '8'],
+      alias: 'Seat',
+      required: false
+    }
+  },
+  '001799': { // Arm Curl
+    seat: {
+      range: ['1', '2', '3', '4', '5', '6', '7', '8', '9'],
+      alias: 'Seat',
+      required: false
+    }
+  },
+  '001999': { // Triceps
+    seat: {
+      range: ['1', '2', '3', '4', '5', '6', '7', '8', '9'],
+      alias: 'Seat',
+      required: false
+    }
+  },
+  '002199': { // Lat Pulldown
+    seat: {
+      range: ['1', '2', '3', '4', '5', '6', '7', '8'],
+      alias: 'Seat',
+      required: false
+    },
+    leftPosition: {
+      range: ['1', '2', '3', '4'],
+      alias: 'Arms',
+      required: false
+    }
+  },
+  '002399': { // Abductor
+    seat: {
+      range: ['1', '2', '3', '4', '5', '6', '7'],
+      alias: 'Back',
+      required: false
+    },
+    leftPosition: {
+      range: ['1', '2', '3', '4', '5', '6'],
+      alias: 'Left Leg',
+      required: false
+    },
+    rightPosition: {
+      range: ['1', '2', '3', '4', '5', '6'],
+      alias: 'Right Leg',
+      required: false
+    }
+  },
+  '002499': { // Adductor
+    seat: {
+      range: ['1', '2', '3', '4', '5', '6', '7'],
+      alias: 'Back',
+      required: false
+    },
+    leftPosition: {
+      range: ['1', '2', '3', '4', '5', '6'],
+      alias: 'Left Leg',
+      required: false
+    },
+    rightPosition: {
+      range: ['1', '2', '3', '4', '5', '6'],
+      alias: 'Right Leg',
+      required: false
+    }
+  },
+  '002699': { // Standing Hip
+    seat: {
+      range: ['1', '2', '3', '4', '5', '6', '7', '8', '9', 'A'],
+      alias: 'Platform',
+      required: false
+    },
+    start: {
+      range: ['1', '2', '3', '4', '5', '6', '7', '8', '9'],
+      alias: 'Start',
+      required: false
+    },
+    leftPosition: {
+      range: ['1', '2', '3', '4', '5', '6', '7'],
+      alias: 'Leg',
+      required: false
+    }
+  },
+  '002899': { /* Lower Back */ },
+  '002999': { /* Seated Calf */ },
+  '003298': { // Runner
+    leftPosition: {
+      range: ['1', '2', '3', '4', '5', '6', '7', '8', '9', '0', 'A', 'C'],
+      alias: 'Chest',
+      required: false
+    }
+  },
+  '003197': { /* Half Rack */ }
+}
+
+export const getAvailableMachineAdjustments = (model: string) => {
+  return new MachineAdjustmentsRecord(machineAdjustments[model as keyof typeof machineAdjustments])
```

Comment:
Why is this better than the last implementation?  Is this protecting against something?

Reply:
The project in typescript won't actually build if you try to index an object typed as a string. You have to give it a type as a keyof some object to index

Reply:
What build issues where you having?

Reply:
<img width="568" alt="Screen Shot 2022-06-29 at 9 07 15 AM" src="https://user-images.githubusercontent.com/18708877/177600139-510af3c0-e6bb-4c39-a208-ea34d314bf57.png">

So it's a semantic error. At the top it says "Element implicitly has an 'any' type because expression of type 'string' can't be used to index type". This is referring to the model number because it is of type `string` which cannot be used to index type `machineAdjustments`

Reply:
![Screen Shot 2022-07-06 at 9 37 25 AM](https://user-images.githubusercontent.com/18708877/177600508-682a71f6-1d6d-48cc-937b-618706d751a2.png)

https://bobbyhadz.com/blog/typescript-type-cannot-be-used-as-an-index-type

This link explains it a little bit more

Reply:
I see.  Thanks for the info.

To make it more type safe then. Let's actually define a `MachineModelType` like so
```
type MachineModelType = keyof typeof machineAdjustments
```
and then make the params into the function be of that type.